packages/alchemy/src/react/hooks/usePlugin.ts

```diff
@@ -0,0 +1,102 @@
+import type { Plugin } from "@alchemy/aa-accounts";
+import { useMutation, type UseMutateFunction } from "@tanstack/react-query";
+import type { Chain, Client, Transport } from "viem";
+import type { SupportedAccount } from "../../config";
+import type { UseSmartAccountClientResult } from "./useSmartAccountClient";
+
+export type UsePluginProps<
+  TTransport extends Transport = Transport,
+  TChain extends Chain | undefined = Chain | undefined,
+  TPlugin extends Plugin = Plugin,
+  TActions extends { [x: string]: (...args: any[]) => unknown } = {
+    [x: string]: (...args: any[]) => unknown;
+  }
+> = {
+  client?: UseSmartAccountClientResult<
+    TTransport,
+    TChain,
+    SupportedAccount<"MultiOwnerModularAccount">
+  >["client"];
+  // TODO: do we really need this? pluginActions does just about everything else we need
+  plugin: TPlugin;
+  pluginActions: (
+    client: Client<
+      TTransport,
+      TChain,
+      SupportedAccount<"MultiOwnerModularAccount">
+    >
+  ) => TActions;
+};
+
+export type UsePluginResult<
+  TActions extends { [x: string]: (...args: any[]) => unknown } = {
+    [x: string]: (...args: any[]) => unknown;
+  }
+> = {
+  executePluginAction: UseMutateFunction<
+    ReturnType<TActions[keyof TActions]>,
+    Error,
+    PluginActionParameters<keyof TActions, TActions>,
+    unknown
+  >;
+  isExecutingPluginAction: boolean;
+};
+
+export type PluginActionParameters<
+  TFunctionName extends keyof TActions,
+  TActions extends { [x: string]: (...args: any[]) => unknown } = {
+    [x: string]: (...args: any[]) => unknown;
+  }
+> = {
+  functionName: TFunctionName;
+  args: Parameters<TActions[TFunctionName]>;
+};
+
+export function usePlugin<
```

Comment:
TODO: jsdocs

Reply:
actually... usePlugin isn't a good name for this hook. This hook is generalizable to using **any** client decorator. Not just the plugin hooks. I'll rename all of this